### PR TITLE
add timeout_us/until to mutex/sem blocking methods

### PR DESCRIPTION
**Pull request by kilograham**

### CODE REVIEW

File: src/common/pico_sync/include/pico/sem.h [hunk 1]
```diff
@@ -90,11 +90,38 @@ void sem_acquire_blocking(semaphore_t *sem);
...
+ * defined timeout has been reached. If the timeout is reached the function will
+ * return false, otherwise it will return true.
+ *
+ * \param sem Pointer to semaphore structure
+ * \param timeout_us Time to wait to acquire the semaphore, in microseconds.
+ * \return false if timeout reached, true if permit was acquired.
+ */
+bool sem_acquire_timeout_us(semaphore_t *sem, uint32_t timeout_us);
+
+/*! \brief Wait for acquire a permit from a semaphore until a specific time
```
Comment (lurch) [inline src/common/pico_sync/include/pico/sem.h:111]: "Wait for acquire" -> "Wait to acquire" ?

File: src/common/pico_sync/include/pico/sem.h [hunk 2]
```diff
@@ -90,11 +90,38 @@ void sem_acquire_blocking(semaphore_t *sem);
...
+ *  \ingroup sem
+ *
+ * This function will block and wait if no permits are available, until the
+ * the specified timeout time. If the timeout is reached the function will
```
Comment (lurch) [inline src/common/pico_sync/include/pico/sem.h:115]: "the the" :wink:

File: src/common/pico_sync/include/pico/sem.h [hunk 3]
```diff
@@ -90,11 +90,38 @@ void sem_acquire_blocking(semaphore_t *sem);
...
+ * return false, otherwise it will return true.
+ *
+ * \param mtx Pointer to mutex structure
```
Comment (lurch) [inline src/common/pico_sync/include/pico/sem.h:118]: mtx -> sem and mutex -> semaphore?

**Review by liamfraser — COMMENTED** [review]
LGTM

**Pull request merged**